GVRf/Framework/framework/src/main/java/org/gearvrf/GVRObjectPicker.java

```diff
@@ -0,0 +1,134 @@
+ * picker is attached to.
+ *
+ * For a {@linkplain GVRSceneObject scene object} to be pickable, it must have a
+ * {@link GVRCollider} component attached to it that is enabled.
+ * The picker returns an array containing all the collisions as instances of GVRPickedObject.
+ * The picked object contains the collider instance, the distance between
+ * the colliding and the scene object and the center of the scene object hit
+ * in world coordinates.
+ *
+ * The picker maintains the list of currently
+ * picked objects which can be obtained with getPicked() and continually
+ * updates it each frame. When a pickable object is inside the view frustum,
+ * the picker generates one or more pick events (IPickEvents interface)
+ * which are sent the event receiver of the scene. These events can be
+ * observed by listeners.
+ *  - onEnter(GVRSceneObject)  called when the scene object enters the frustum.
+ *  - onExit(GVRSceneObject)   called when the scene object exits the frustum.
+ *  - onInside(GVRSceneObject) called while the scene object is inside the frustum.
+ *  - onPick(GVRPicker)        called when the set of picked objects changes.
+ *  - onNoPick(GVRPicker)      called once when nothing is picked.
+ *
+ * @see IPickEvents
+ * @see GVRSceneObject.attachComponent
+ * @see GVRCollider
+ * @see GVRComponent.setEnable
+ * @see GVRPickedObject
+ * @see GVRPicker
+ */
+public class GVRObjectPicker extends GVRPicker
+{
+
+    /**
+     * Construct a picker which picks from a given scene.
+     * @param context context that owns the scene
+     * @param scene scene containing the scene objects to pick from
+     */
+    public GVRObjectPicker(GVRContext context, GVRScene scene)
+    {
+        super(context, scene);
+    }
+
+    public void onDrawFrame(float frameTime)
+    {
+        if (isEnabled())
+        {
+            doPick();
+        }
+    }
+
+    /**
+     * Scans the scene graph to collect picked items
+     * and generates appropriate pick events.
+     * This function is called automatically by
+     * the picker if it is attached to a scene object.
+     * You can instantiate the picker and not attach
+     * it to a scene object. In this case you must
+     * manually set the pick ray and call doPick()
+     * to generate the pick events.
+     * @see IPickEvents
+     * @see GVRFrustumPicker.pickVisible
+     */
+    public void doPick()
+    {
+        GVRSceneObject owner = getOwnerObject();
+        GVRPickedObject[] picked = GVRFrustumPicker.pickVisible(mScene);
+
+        if (owner != null)
+        {
+            GVRSceneObject.BoundingVolume bv1 = owner.getBoundingVolume();
+
+            for (int i = 0; i < picked.length; ++i)
+            {
+                GVRPickedObject hit = picked[i];
+
+                if (hit != null)
+                {
+                    GVRSceneObject sceneObj = hit.hitObject;
+                    GVRSceneObject.BoundingVolume bv2 = sceneObj.getBoundingVolume();
```

Comment:
Might be beneficial if getBoundingVolume was updated not to always create a new Java object.